runtime.go

```diff
@@ -74,7 +74,7 @@ func contextualMain(main func(ctx context.Context, args []string, logger golog.L
 	}
 }
 
-var fatal = func(logger golog.Logger, args ...interface{}) {
+var fatal = func(logger ILogger, args ...interface{}) {
```

Comment:
This exists because contextual main required a golog logger which isn't an interface. We can now in theory clean this up by having the tests create their own logger type that embeds a golog.Logger, but replaces the `Fatal` method.

But I'm not confident this is as simple as I think it will be and the contextual main change is blocking more important work.